src/main/java/domain/AttendanceFactory.java

```diff
@@ -0,0 +1,55 @@
+package domain;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class AttendanceFactory {
+    
+    private static final Map<String, List<Attendance>> attendances = new HashMap<>();
+    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+    
+    public static AttendanceBook createAttendanceBook() throws IOException {
+        try (
+                FileReader fileReader = new FileReader("./src/main/resources/attendances.csv");
+                BufferedReader reader = new BufferedReader(fileReader)
+        ) {
+            String str = reader.readLine();
+            while ((str = reader.readLine()) != null) {
+                insertAttendance(str);
+            }
+            return buildAttendanceBook();
+        }
+    }
+    
```

Comment:
attendances를 static field로 관리할 필요가 있을까요?
여러 번 createAttendanceBook()을 호출하면 이전 데이터를 유지한 채 계속 추가될 가능성이 있습니다.
물론 지금 프로그램에서는 한번만 호출되고 있지만 이 내용을 알고 계시면 좋을 것 같아서 말씀드립니다.
```suggestion
    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
    
    public static AttendanceBook createAttendanceBook() throws IOException {
        Map<String, List<Attendance>> attendances = new HashMap<>();
        try (
                FileReader fileReader = new FileReader("./src/main/resources/attendances.csv");
                BufferedReader reader = new BufferedReader(fileReader)
        ) {
            String str = reader.readLine();
            while ((str = reader.readLine()) != null) {
                insertAttendance(attendances, str);
            }
            return buildAttendanceBook(attendances);
        }
    }
    
```

Reply:
고마워요 앤지 ! 그런 상황이 올 수 있겠군요. 앞으로 구현할때 다양한 상황을 고려해보도록 하겠습니다 😊